wpilibj/src/main/java/edu/wpi/first/wpilibj/XboxController.java

```diff
@@ -93,170 +108,136 @@ public boolean getBumper(Hand hand) {
+  @Deprecated
   public boolean getYButton() {
     return getRawButton(4);
   }
 
   /**
-   * Read the value of the stick button on the controller.
-   *
-   * @param hand Side of controller whose value should be returned.
-   * @return The state of the button.
-   */
-  @Override
-  public boolean getStickButton(Hand hand) {
-    if (hand.equals(Hand.kLeft)) {
-      return getRawButton(9);
-    } else {
-      return getRawButton(10);
-    }
-  }
-
-  /**
    * Read the value of the back button on the controller.
    *
+   * @deprecated Use getButton() instead.
    * @return The state of the button.
    */
+  @Deprecated
   public boolean getBackButton() {
     return getRawButton(7);
   }
 
   /**
    * Read the value of the start button on the controller.
    *
+   * @deprecated Use getButton() instead.
    * @return The state of the button.
    */
+  @Deprecated
   public boolean getStartButton() {
     return getRawButton(8);
   }
 
-  @Override
-  public int getPOV(int pov) {
-    return m_ds.getStickPOV(getPort(), pov);
-  }
-
-  @Override
-  public int getPOVCount() {
-    return m_ds.getStickPOVCount(getPort());
-  }
-
-  @Override
-  public HIDType getType() {
-    return HIDType.values()[m_ds.getJoystickType(getPort())];
-  }
-
-  @Override
-  public String getName() {
-    return m_ds.getJoystickName(getPort());
-  }
-
-  @Override
-  public void setOutput(int outputNumber, boolean value) {
-    m_outputs = (m_outputs & ~(1 << (outputNumber - 1))) | ((value ? 1 : 0) << (outputNumber - 1));
-    HAL.setJoystickOutputs((byte) getPort(), m_outputs, m_leftRumble, m_rightRumble);
+  /**
+   * Whether the button was pressed since the last check. Button indexes begin at 1. The button
+   * type is mapped to a button index first.
+   *
+   * @param button The button index, beginning at 1.
+   * @return Whether the button was pressed since the last check.
+   */
+  boolean getButtonPressed(Button button) {
+    return getRawButtonPressed(getButtonIndex(button));
   }
 
-  @Override
-  public void setOutputs(int value) {
-    m_outputs = value;
-    HAL.setJoystickOutputs((byte) getPort(), m_outputs, m_leftRumble, m_rightRumble);
+  /**
+   * Whether the button was released since the last check. Button indexes begin at 1. The button
+   * type is mapped to a button index first.
+   *
+   * @param button The button index, beginning at 1.
+   * @return Whether the button was released since the last check.
+   */
+  boolean getButtonReleased(Button button) {
+    return getRawButtonReleased(getButtonIndex(button));
   }
 
-  @Override
-  public void setRumble(RumbleType type, double value) {
-    if (value < 0) {
-      value = 0;
-    } else if (value > 1) {
-      value = 1;
-    }
-    if (type == RumbleType.kLeftRumble) {
-      m_leftRumble = (short) (value * 65535);
-    } else {
-      m_rightRumble = (short) (value * 65535);
+  /**
+   * Returns button index corresponding to a button type.
+   */
+  int getButtonIndex(Button button) {
```

Comment:
public?

Reply:
I was actually considering making this private because I couldn't think of a valid use case for our users.